src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.logic.commands.SearchCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new SearchCommand object
+ */
+public class SearchCommandParser implements Parser<SearchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SearchCommand
+     * and returns a SearchCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+        try {
+            String moduleCode = StringUtil.ignoreCase(trimmedArgs);
+            return new SearchCommand(moduleCode);
+        } catch (IllegalArgumentException e) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    SearchCommand.MESSAGE_USAGE));
+        }
```

Comment:
I think there's no `IllegalArgumentException` to catch here LOL `StringUtil.ignoreCase` and `SearchCommand's` constructor both don't throw ittt

Reply:
This is with regards to Yan's comment because the command `search MA 1521` will not throw error when it is supposed to but after the IllegalArgumentException was added it was able to catch the error

Reply:
Oooh okay! My bad hehe I didn't realised that `checkArgument` in `StringUtil.ignoreCase()` throws an undeclared `IllegalArgumentException` (since its a runtime error)

Reply:
No need to change, this is great :D